src/site/content/en/metrics/fcp/index.md

```diff
@@ -0,0 +1,132 @@
+authors:
+  - philipwalton
+date: 2019-11-07
+updated: 2019-11-07
+description: |
+  This post introduces the First Contentful Paint (FCP) metric and explains
+  how to measure it
+tags:
+  - performance
+  - metrics
+---
+
+{% Aside %}
+  First Contentful Paint (FCP) is an important, user-centric metric for
+  measuring [perceived load speed](/user-centric-metrics/#types-of-metrics)
+  because it marks the first point in the page load timeline that the user can
+  see anything on the screen—a fast FCP helps reassure the user that something
+  is [happening](/user-centric-metric/#user-centric-metric-questions).
+{% endAside %}
+
+## What is FCP?
+
+The First Contentful Paint (FCP) metric measures the time from when the page
+starts loading to when any part of the page's content is rendered on the screen.
+For this metric, "content" refers to text, images (including background images),
+`<svg>` elements, or non-white `<canvas>` elements.
+
+[![FCP timeline from google.com](fcp-filmstrip.png)](fcp-filmstrip.png)
+
+In the above load timeline, the first contentful paint happens in the second
+frame, as that's when the first text and image elements are rendered to the
+screen.
+
+You'll notice that though some of the content has rendered, not all of it has
+rendered. This is an important distinction to make between _First_ Contentful
+Paint (FCP) and _[Largest Contentful Paint](/largest-contentful-paint/)_
+(LCP)—which aims to measure when the page's main contents have finished loading.
+
+## How to measure FCP
+
+FCP can be measured [in the lab](/user-centric-metrics/#in-the-lab) or [in the
+field](/user-centric-metrics/#in-the-field), and it's available in the following
+tools:
+
+### Lab tools
+
+- [Lighthouse](https://developers.google.com/web/tools/lighthouse/)
+- [WebPageTest](https://www.webpagetest.org/)
+- [Chrome DevTools](https://developers.google.com/web/tools/chrome-devtools/)
+- [PageSpeed Insights](https://developers.google.com/speed/pagespeed/insights/)
+
+### Field tools
+
+- [PageSpeed Insights](https://developers.google.com/speed/pagespeed/insights/)
+- [Chrome User Experience
+  Report](https://developers.google.com/web/tools/chrome-user-experience-report)
+- [Firebase Performance
+  Monitoring](https://firebase.google.com/docs/perf-mon/get-started-web) (beta)
+
+### Measure FCP in JavaScript
+
+You can measure FCP in JavaScript using the [Paint Timing
+API](https://w3c.github.io/paint-timing/). The following example shows how to
+create a
+[`PerformanceObserver`](https://developer.mozilla.org/en-US/docs/Web/API/PerformanceObserver)
+that listens for paint timing entries and logs the `first-contentful-paint` time
+to the console:
+
+```js
+// Create the Performance Observer instance.
+const observer = new PerformanceObserver((list) => {
+  for (const entry of list.getEntriesByName('first-contentful-paint')) {
+    // Log the value of FCP to the console.
+    console.log('FCP:', entry.startTime);
```

Comment:
Perhaps you can disconnect the observer when you enter the for loop (there is only one FCP).

Reply:
Good call.